src/parser.cc

```diff
@@ -1420,15 +1502,18 @@ bool readXml(tinyxml2::XMLElement *_xml, ElementPtr _sdf,
       {
         if (!_sdf->HasElement(elemDesc->GetName()))
         {
+          const std::string elemXmlPath = _sdf->XmlPath() + "/" +
+              elemDesc->GetName();
           if (_sdf->GetName() == "joint" &&
               _sdf->Get<std::string>("type") != "ball")
           {
-            _errors.push_back({
+            Error err(
                 ErrorCode::ELEMENT_MISSING,
                 "XML Missing required element[" + elemDesc->GetName() +
                 "], child of element[" + _sdf->GetName() + "]",
-                sourcePath,
-                elemXml->GetLineNum()});
+                sourcePath, elemXml->GetLineNum());
+            err.SetXmlPath(elemXmlPath);
```

Comment:
There are many areas where `err.SetXmlPath` is called right after `Error` is constructed, could we create another constructor that takes in the `elemXmlPath`?

Reply:
Creating another constructor breaks ABI, as there are cases where an `Error` only has the XML path, which collides with the old API of only having a file path (both have `std::string` as arguments), there was a discussion about this with @azeey in second resolved change.

Reply:
I see, thanks for clarifying!